gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/ConnectionPool.java

```diff
@@ -18,476 +18,81 @@
-                return leastUsedConn;
-            }
-        }
-    }
-
-    public void returnConnection(final Connection connection) throws ConnectionException {
-        logger.debug("Attempting to return {} on {}", connection, host);
-        if (isClosed()) throw new ConnectionException(host.getHostUri(), host.getAddress(), "Pool is shutdown");
-
-        final int borrowed = connection.borrowed.decrementAndGet();
-
-        if (connection.isDead()) {
-            logger.debug("Marking {} as dead", this.host);
-            this.replaceConnection(connection);
-        } else {
-            if (bin.contains(connection) && borrowed == 0) {
-                logger.debug("{} is already in the bin and it has no inflight requests so it is safe to close", connection);
-                if (bin.remove(connection))
-                    connection.closeAsync();
-                return;
-            }
-
-            // destroy a connection that exceeds the minimum pool size - it does not have the right to live if it
-            // isn't busy. replace a connection that has a low available in process count which likely means that
-            // it's backing up with requests that might never have returned. consider the maxPoolSize in this condition
-            // because if it is equal to 1 (which it is for a session) then there is no need to replace the connection
-            // as it will be responsible for every single request. if neither of these scenarios are met then let the
-            // world know the connection is available.
-            final int poolSize = connections.size();
-            final int availableInProcess = connection.availableInProcess();
-            if (poolSize > minPoolSize && borrowed <= minSimultaneousUsagePerConnection) {
-                if (logger.isDebugEnabled())
-                    logger.debug("On {} pool size of {} > minPoolSize {} and borrowed of {} <= minSimultaneousUsagePerConnection {} so destroy {}",
-                            host, poolSize, minPoolSize, borrowed, minSimultaneousUsagePerConnection, connection.getConnectionInfo());
-                destroyConnection(connection);
-            } else if (availableInProcess < minInProcess && maxPoolSize > 1) {
-                if (logger.isDebugEnabled())
-                    logger.debug("On {} availableInProcess {} < minInProcess {} so replace {}", host, availableInProcess, minInProcess, connection.getConnectionInfo());
-                replaceConnection(connection);
-            } else
-                announceAvailableConnection();
-        }
-    }
-
-    Client getClient() {
-        return client;
-    }
-
-    Cluster getCluster() {
-        return cluster;
-    }
-
-    public boolean isClosed() {
-        return closeFuture.get() != null;
-    }
-
+public interface ConnectionPool {
+    int DEFAULT_MAX_POOL_SIZE = 8;
     /**
-     * Permanently kills the pool.
+     * Since 3.4.3 only max pool size is used and this setting is ignored.
```

Comment:
just another place where the standard javadoc deprecation message needs to be used.